tests_e2e/tests/scripts/agent_firewall-verify_all_firewall_rules.py

```diff
@@ -0,0 +1,402 @@
+#!/usr/bin/env pypy3
+
+# Microsoft Azure Linux Agent
+#
+# Copyright 2018 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# This script checks all agent firewall rules added properly and working as expected
+#
+import argparse
+import logging
+import os
+import pwd
+import socket
+import subprocess
+import sys
+import time
+import traceback
+from typing import List
+
+if sys.version_info[0] == 3:
+    import http.client as httpclient
+elif sys.version_info[0] == 2:
+    import httplib as httpclient  # pylint: disable=E0401
+
+NON_ROOT_USER = 'unknown_user'
+ROOT_USER = 'root'
+WIRESERVER_ENDPOINT_FILE = '/var/lib/waagent/WireServerEndpoint'
+WIRESERVER_URL = '168.63.129.16'
+VERSIONS_PATH = '/?comp=versions'
+
+
+def get_wireserver_ip():
+    try:
+        with open(WIRESERVER_ENDPOINT_FILE, 'r') as f:
+            wireserver_ip = f.read()
+    except Exception:
+        wireserver_ip = WIRESERVER_URL
+    return wireserver_ip
+
+
+def switch_user(user):
+    # This function switches the function to a given user
+    try:
+        uid = pwd.getpwnam(user)[2]
+        logging.info("uid:%s and user name:%s", uid, user)
+        os.seteuid(uid)
+    except Exception as e:
+        raise Exception("Error -- failed to switch user to {0} : Failed with exception {1}".format(user, e))
+
+
+def get_root_accept_rule(command: str):
+    return ['sudo', 'iptables', '-t', 'security', command, 'OUTPUT', '-d', get_wireserver_ip(), '-p', 'tcp', '-m',
+            'owner',
+            '--uid-owner',
+            '0', '-j', 'ACCEPT', '-w']
+
+
+def get_non_root_accept_rule(command: str):
+    return ['sudo', 'iptables', '-t', 'security', command, 'OUTPUT', '-d', get_wireserver_ip(), '-p', 'tcp',
+            '--destination-port', '53', '-j',
+            'ACCEPT', '-w']
+
+
+def get_non_root_drop_rule(command: str):
+    return ['sudo', 'iptables', '-t', 'security', command, 'OUTPUT', '-d', get_wireserver_ip(), '-p', 'tcp', '-m',
+            'conntrack', '--ctstate',
+            'INVALID,NEW', '-j', 'DROP', '-w']
+
+
+def execute_cmd(cmd: List[str]):
```

Comment:
use shellutil.run_command

Reply:
I need retuncode and stderr too. The shullutil don't return those

Reply:
i think you missed this comment in your last update

Reply:
I addressed right, I have used sheluti.run_command